src/main.ts

```diff
@@ -1,74 +1,104 @@
-import { app, BrowserWindow, Menu, ipcMain } from 'electron';
+import { spawn, ChildProcess } from 'child_process';
+import { app, BrowserWindow, ipcMain } from 'electron';
+import { createTopMenu } from './main/menu';
+import { pathToFileURL } from 'url';
+import { configMaestro } from './utils/config';
+import { setSessionId, getSessionId } from './cosimulation/simulationContext';
+
 import * as path from 'path';
+import * as fs from 'fs';
+import { exec } from 'node:child_process';
+import kill from 'tree-kill';
+
 let mainWindow: BrowserWindow | null = null;
+let maestroProcess: ChildProcess | null = null;
+const { coeJarPath, simulationConfigPath, fmusPath, multiModels, outputPath } = configMaestro;
+
+if (!coeJarPath || !simulationConfigPath || !fmusPath || !multiModels || !outputPath) {
+  console.error('Missing required paths in config.json');
+  process.exit(1);
+}
+
+const isDev = process.env.NODE_ENV === 'development';
+process.env.NODE_ENV =
+  process.env.NODE_ENV || (isDev ? 'development' : 'production');
+
+const preloadPath = isDev
+  ? path.resolve(__dirname, 'preload.js')
+  : path.resolve(app.getAppPath(), 'dist/preload.js');
+
+const startUrl = isDev
+  ? 'http://localhost:3000'
+  : `file://${path.resolve(app.getAppPath(), 'dist/index.html')}`;
+
+const iconPath = isDev
+  ? path.resolve(__dirname, 'resources/into-cps/appicon/into-cps-logo.png.ico')
+  : path.resolve(
+      app.getAppPath(),
+      'dist/resources/into-cps/appicon/into-cps-logo.png.ico',
+    );
+
+const maestroJarPath = isDev
+  ? path.resolve(__dirname, 'resources/maestro/maestro.jar')
+  : path.resolve(app.getAppPath(), 'dist/resources/maestro/maestro.jar');
+
+const tempMaestroJarPath = path.join(app.getPath('temp'), 'maestro.jar');
+
+function extractMaestroJar() {
+  if (!fs.existsSync(maestroJarPath)) {
+    console.error(`Maestro JAR not found at ${maestroJarPath}`);
+    throw new Error(`Maestro JAR not found at ${maestroJarPath}`);
+  } 
+
+  if(!fs.existsSync(tempMaestroJarPath)){
+    fs.copyFileSync(maestroJarPath, tempMaestroJarPath);
+  } else {
+    console.log(`Maestro JAR already extracted to ${tempMaestroJarPath}`);
+  }
+}
+
+function checkJavaInstallation(): Promise<boolean> {
```

Comment:
This is a rabbit hole. Even if the java is installed, the java class paths could be wrong. I am thinking aloud here. Is it sufficient to start maestro and then check if it exits for any reason. The reason could be used instead of checking the underlying problems.
Again, there might be value in providing this hint to the user. You please decide either way.

Reply:
I’ve adjusted the approach to focus on monitoring Maestro's behavior during its startup process rather than directly validating Java installation or classpath configurations, obderving Maestro's standard output and error streams to identify issues dynamically.